### PR TITLE
fix(core): fix database stuck on startup after some table drops

### PR DESCRIPTION
**Pull request by ideoma**
Some fuzz tests in CI were stuck because logging exceptions did not return the logging sequence.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/tasks/ColumnPurgeTask.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeOperator.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/tasks/ColumnPurgeTask.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeOperator.java|3|3|100.00%|

File: core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java [hunk 1]
```diff
@@ -231,7 +231,7 @@ private void processTableRecords(CairoEngine engine) {
                     count++;
                     long ts = rec.getTimestamp(0);
                     if (ts != lastTs || taskRun == null) {
-                        if (taskRun != null) {
+                        if (taskRun != null && !taskRun.isEmpty()) {
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java:234]: isn't this leaking `taskRun` objects? 
say `taskRun` is not null, but it's empty. (=no name set, because token was null before) 
 
this means you had previously `pop()`-ed  from the pool. and now you take another one, without returning the original back. 

am I missing anything?

Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java]: Good spot, fixed

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/tasks/ColumnPurgeTask.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeOperator.java|3|3|100.00%|

File: core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java [hunk 2]
```diff
@@ -232,7 +232,9 @@ private void processTableRecords(CairoEngine engine) {
                     long ts = rec.getTimestamp(0);
                     if (ts != lastTs || taskRun == null) {
                         if (taskRun != null) {
-                            columnPurgeOperator.purgeExclusive(taskRun);
+                            if (!taskRun.isEmpty()) {
+                                columnPurgeOperator.purgeExclusive(taskRun);
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java:236]: One more scenario I am not sure about: 

1. You just successfully purged a table. 
2. In the next iteration, you pick a table which happens to be dropped. Thus the guard on line no. 253 will short-circuit the loop. 
3. The dropped table was the last record -> `records.hasNext()` returns false and you exit the loop. 
4. At this point the `taskRun` is not null and it's set to the table already purged in step no. 1.
5. So the line no. 276 will purge it again!

In other words: If the last table is dropped then the last non-dropped table will be purged twice. I am not sure if that may cause a problem (probably not), just highlighting it for your consideration.

**Review by jerrinot — DISMISSED** [review]
feel free to resolve my comment about double-purging. I am assuming purge is idempotent, I just wanted to make sure you are aware of this possibility.

File: core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java (hunk 2 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java]: I changed the code to avoid double runs, thanks

**Review by jerrinot — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/tasks/ColumnPurgeTask.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeOperator.java|3|3|100.00%|
